### DISCUSSION TITLE
`findItems` query - Input Parameters

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#88] [General]
Hi @ralphhanna,

Consider the below example,

![image](https://user-images.githubusercontent.com/55505611/231732763-bf76cecf-a5e7-4dc9-b7f8-a082f2c73145.png)

Here, the Receive Task `RT1` has an input parmater named `sender` with a default value.

In DB, the record associated with the above task follows the below structure:

![image](https://user-images.githubusercontent.com/55505611/231733397-c41750f6-bdfe-424c-9129-a0b00473d37e.png)

To find a item, normally we would use `server.datastore.findItems(query)` while the `query` can be anything as long as it fits into the record structure of the task (the one above).

But I don't see the input parameters of a task recorded into the item's record in DB. 

If I want to query a task whose input parameter matches a value, 
Something like, `server.datastore.findItems({"items.status": "wait", "items.inputParams.sender": "<something>"})`.

I presume I couldn't do that now as the input parameters aren' there! 

- Why the input paramters are not part of the record in DB?
- How to query an item based on input parameters too?

Thank you!

**Comment by ralphhanna** [comment]
In Release 1.3.22 added **item.vars** to store any variables related to the item as follows

## Inside your service:
```js
  async service1(input, context) {
  ...
  item.vars= input;
 ...
}
```
## MongoDB:
As a result MongoDB stores item.vars
![image](https://github.com/ralphhanna/bpmn-server/assets/11893416/320e2e2f-e6e3-46a9-964b-f10e91ce8a32)

## Using findItems:
```js
            query = { "data.caseId": caseId, "items.vars.param1": 'value1' };
            items = await server.dataStore.findItems(query);
            console.log('items count',items);

```
